### PR TITLE
[accept-language-parser] Allow passing `null` to pick

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
`pick` accepts any falsy values for its parameters: https://github.com/opentable/accept-language-parser/blob/3a1133a42889482e768f01d0b4e4384bb14aa762/index.js#L34

In general passing a falsy value is probably a bug. However, since `Headers#get` returns `string | null` (https://github.com/microsoft/TypeScript/blob/main/lib/lib.dom.d.ts#L8877), it's handy to allow `null` as `acceptedLanguage` so that this pattern is accepted:

```js
pick(supportedLanguages, headers.get("accepted-language"));
```

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/opentable/accept-language-parser/blob/3a1133a42889482e768f01d0b4e4384bb14aa762/index.js#L34
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
Also cc @matteofigus @federicomaffei. It looks like you used to maintain this package, so maybe you have opinions about its type definitions.

File: types/accept-language-parser/index.d.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@
 export function parse(acceptLanguage?: string): Language[];
 export function pick<T extends string>(
     supportedLanguages: T[],
-    acceptLanguage: string | Language[],
+    acceptLanguage: string | Language[] | null,
```
Comment (peterblazejewicz) [inline types/accept-language-parser/index.d.ts:13]:
```ts
    acceptLanguage: string | Language[] | null,
```
why this change? Docs (readme) and tests, never expect anything but string here. Source code works with anything that evaluates to false, but this is hardly expect to be advertised in docs, so why 'null' as very specific type in JS?

Comment (nicolo-ribaudo) [reply types/accept-language-parser/index.d.ts:13]: From https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61235#issue-1304026307:

> `pick` accepts any falsy values for its parameters: https://github.com/opentable/accept-language-parser/blob/3a1133a42889482e768f01d0b4e4384bb14aa762/index.js#L34
>
>In general passing a falsy value is probably a bug. However, since `Headers#get` returns `string | null` (https://github.com/microsoft/TypeScript/blob/main/lib/lib.dom.d.ts#L8877), it's handy to allow `null` as `acceptedLanguage` so that this pattern is accepted:
>
> ```js
> pick(supportedLanguages, headers.get("accepted-language"));
> ```

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
Its not clear if this change is desired and follows the expectation of the usage.. We need feedback from other authors of this d.ts to be sure.

**Comment by niklas-sparfeld-gcx** [comment]
I guess I am the original author and I am sorry to keep you waiting.

All I can say is that I have not worked in TypeScript for approximately 5 years now ;-)

So I cannot really have a very helpful opinion on this topic.

---

But here it is: My opinion :-)

If I were the author of accept-language-parser in another language like Kotlin, I would *not* have accepted `null` as input. 

But since the `accept-language-parser` code in fact *does* accept and handle it, I would change the type definitions accordingly. The library authors chose to accept null, so the types should as well.
